src/inpod/protocol.rs

```diff
@@ -280,6 +280,35 @@ mod tests {
         assert!(matches!(m, WorkloadMessage::AddWorkload(_)));
     }
 
+    #[test]
+    fn test_parse_add_workload_with_info() {
+        let owned_fd: OwnedFd = std::fs::File::open("/dev/null").unwrap().into();
```

Comment:
Deferrable nit: would rather use something like [`tempfile` ](https://docs.rs/tempfile/latest/tempfile/) to generate a junk file to get FDs off of, so we have a guarantee the underlying file goes away no matter what.

Reply:
`/dev/null` is pretty much a constant - I don't need it to be deleted.. I'm not sure i understand the advantage of using a tempfile here?

Reply:
It's not a big deal - I'm just thinking that if we ever have to debug tests or suites leaking FDs, it's _much_ easier to find FDs dangling from deleted one-per-test tempfiles than from `/dev/null` which will always be there and which potentially many other things on the system are opening FDs against and using.